mlrun/run.py

```diff
@@ -1218,3 +1226,614 @@ def wait_for_runs_completion(runs: list, sleep=3, timeout=0, silent=False):
+            :param data_item: The `DataItem` to parse.
+
+            :returns: The `DataItem` as a `pandas.DataFrame`.
+            """
+            return data_item.as_df()
+
+        @staticmethod
+        def parse_numpy_array(data_item: DataItem) -> np.ndarray:
+            """
+            Parse an MLRun `DataItem` to a `numpy.ndarray`.
+
+            :param data_item: The `DataItem` to parse.
+
+            :returns: The `DataItem` as a `numpy.ndarray`.
+            """
+            return data_item.as_df().to_numpy()
+
+        @staticmethod
+        def parse_dict(data_item: DataItem) -> dict:
+            """
+            Parse an MLRun `DataItem` to a `dict`.
+
+            :param data_item: The `DataItem` to parse.
+
+            :returns: The `DataItem` as a `dict`.
+            """
+            return data_item.as_df().to_dict()
+
+        @staticmethod
+        def parse_list(data_item: DataItem) -> list:
+            """
+            Parse an MLRun `DataItem` to a `list`.
+
+            :param data_item: The `DataItem` to parse.
+
+            :returns: The `DataItem` as a `list`.
+            """
+            return data_item.as_df().to_numpy().tolist()
+
+        @staticmethod
+        def parse_object(data_item: DataItem) -> object:
+            """
+            Parse an MLRun `DataItem` to its unpickled object. The pickle file will be downloaded to a local temp
+            directory and then loaded.
+
+            :param data_item: The `DataItem` to parse.
+
+            :returns: The `DataItem` as the original object that was pickled once it was logged.
+            """
+            object_file = data_item.local()
+            with open(object_file, "rb") as pickle_file:
+                obj = cloudpickle.load(pickle_file)
+            return obj
+
+    # The map to use for parsing an object by its type:
+    PARSER_MAP = {
+        str(pd.DataFrame): _Parser.parse_pandas_dataframe,
+        str(np.ndarray): _Parser.parse_numpy_array,
+        str(dict): _Parser.parse_dict,
+        str(list): _Parser.parse_list,
+        str(object): _Parser.parse_object,
+    }
+
+    def __init__(self):
+        """
+        Initialize a context handler.
+        """
+        self._context = None  # type: MLClientCtx
+
+    def look_for_context(self, args: tuple, kwargs: dict):
+        """
+        Look for an MLRun context (`mlrun.MLClientCtx`). The handler will look for a context in the given order:
+          1. The given arguments.
+          2. The given keyword arguments.
+          3. If an MLRun RunTime was used the context will be located via the `mlrun.get_or_create_ctx` method.
+
+        :param args:   The arguments tuple passed to the function.
+        :param kwargs: The keyword arguments dictionary passed to the function.
+        """
+        # Search in the given arguments:
+        for argument in args:
+            if isinstance(argument, MLClientCtx):
+                self._context = argument
+                return
```

Comment:
I think you should verify before whether args in not None

Reply:
As far as I know, the `*args` cannot be None, only an empty tuple, same for `**kwargs`, only an empty dictionary. 
![image](https://user-images.githubusercontent.com/83535508/185411311-3552b2de-17b0-4258-9e7e-9804454e2914.png)